File: src/SOP8/Base/Base.Model/Sop/Component/GridColumn.cs
using System;
using dnsDapperDBUtil.DataAccessLayer.IDAL;

namespace Base.Model.Sop.Component
{
	public class GridColumn : Table
	{
		public enum Fields { grid_sn, column_no, width };
		public enum WriteFields { grid_sn, column_no, width };

		public int grid_sn { get; set; }
		public int column_no { get; set; }
		public int width { get; set; }

		public static string TableName { get { return "so_compn_grid_column"; } }

		public override string GetTableName()
		{
			return TableName;
		}

		public override string GetPrimaryCondition()
		{
			return string.Format("{0} = {1} and {2} = {3}", Fields.grid_sn, grid_sn, Fields.column_no, column_no);
		}

		public override Type GetFieldType()
		{
			return typeof(Fields);
		}

		public override Type GetWriteFieldType()
		{
			return typeof(WriteFields);
		}

		public void FromCopy(GridColumn obj)
		{
			this.grid_sn = obj.grid_sn;
			this.column_no = obj.column_no;
			this.width = obj.width;
		}
	}
}

File: src/SOP8/Egovframework/Base/base-model/src/main/java/egovframework/base/model/sop/component/GridColumn.java
package egovframework.base.model.sop.component;

import lombok.*;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GridColumn {
	public enum Fields { grid_sn, column_no, width }

	private int grid_sn;

	private int column_no;

	private int width;

	public static final String autoIncreaseField = null;
	public static final String tableName = "so_compn_grid_column";
	public static final String[] primaryKeys = { "grid_sn", "column_no" };

}
